homeassistant/helpers/translation.py

```diff
@@ -348,25 +371,168 @@ async def async_get_translations(
 ) -> dict[str, str]:
     """Return all backend translations.
 
-    If integration specified, load it for that one.
-    Otherwise default to loaded integrations combined with config flow
+    If integration is specified, load it for that one.
+    Otherwise, default to loaded integrations combined with config flow
     integrations if config_flow is true.
     """
+    if integrations is None and config_flow:
+        components = (await async_get_config_flows(hass)) - hass.config.components
+    else:
+        components = _async_get_components(hass, category, integrations)
+
+    if TRANSLATION_FLATTEN_CACHE in hass.data:
+        cache: _TranslationCache = hass.data[TRANSLATION_FLATTEN_CACHE]
+    else:
+        cache = hass.data[TRANSLATION_FLATTEN_CACHE] = _TranslationCache(hass)
```

Comment:
We don't need these checks since we call `async_setup` from bootstrap.
```suggestion
    cache: _TranslationCache = hass.data[TRANSLATION_FLATTEN_CACHE]
```